bench_test.go

```diff
@@ -54,41 +108,126 @@ func BenchmarkParallelGetSet(b *B) {
 		})
 	}
 
-	radix, err := NewPool("tcp", "127.0.0.1:6379", parallel)
-	if err != nil {
-		b.Fatal(err)
-	}
 	b.Run("radix", func(b *B) {
-		do(b, func() {
-			err := radix.Do(WithConn("foo", func(conn Conn) error {
-				if err := conn.Do(Cmd(nil, "SET", "foo", "bar")); err != nil {
-					return err
+		b.Run("no pipelining", func(b *B) {
+			radix, err := NewPool("tcp", "127.0.0.1:6379", parallel, PoolPipelineWindow(0, 0))
+			if err != nil {
+				b.Fatal(err)
+			}
+			defer radix.Close()
+			<-radix.initDone
+
+			do(b, func() {
+				err := radix.Do(WithConn("foo", func(conn Conn) error {
+					if err := conn.Do(Cmd(nil, "SET", "foo", "bar")); err != nil {
+						return err
+					}
+					var out string
+					if err := conn.Do(Cmd(&out, "GET", "foo")); err != nil {
+						return err
+					} else if out != "bar" {
+						b.Fatal("got wrong value")
+					}
+					return nil
+				}))
+				if err != nil {
+					b.Fatal(err)
+				}
+			})
+		})
+
+		b.Run("one pipeline", func(b *B) {
+			radix, err := NewPool("tcp", "127.0.0.1:6379", parallel, PoolPipelineConcurrency(1))
+			if err != nil {
+				b.Fatal(err)
+			}
+			defer radix.Close()
+			<-radix.initDone
+
+			do(b, func() {
+				if err := radix.Do(Cmd(nil, "SET", "foo", "bar")); err != nil {
+					b.Fatal(err)
 				}
 				var out string
-				if err := conn.Do(Cmd(&out, "GET", "foo")); err != nil {
-					return err
+				if err := radix.Do(Cmd(&out, "GET", "foo")); err != nil {
+					b.Fatal(err)
+				} else if out != "bar" {
+					b.Fatal("got wrong value")
 				}
-				return nil
-			}))
+			})
+		})
+
+		b.Run("multiple pipelines", func(b *B) {
```

Comment:
For the sake of looking good, let's call this `"default"` :P

Reply:
Done